src/Symfony/Component/Lock/Store/CombinedStore.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\Store;
+
+use Symfony\Component\Lock\Exception\InvalidArgumentException;
+use Symfony\Component\Lock\Exception\LockConflictedException;
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\QuorumInterface;
+use Symfony\Component\Lock\StoreInterface;
+
+/**
+ * CombinedStore is a StoreInterface implementation able to manage and synchronize several StoreInterfaces.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+class CombinedStore implements StoreInterface, ExpirableStoreInterface
+{
+    /** @var StoreInterface[] */
+    private $stores;
+    /** @var QuorumInterface */
+    private $quorum;
+
+    /**
+     * @param StoreInterface[] $stores The decorated stores
+     * @param QuorumInterface  $quorum
+     *
+     * @throws InvalidArgumentException
+     */
+    public function __construct(array $stores, QuorumInterface $quorum)
+    {
+        foreach ($stores as $store) {
+            if (!$store instanceof StoreInterface) {
+                throw new InvalidArgumentException(
+                    sprintf('The store must implement "%s". Got "%s".', StoreInterface::class, get_class($store))
+                );
+            }
+        }
+
+        $this->stores = $stores;
+        $this->quorum = $quorum;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function save(Key $key)
+    {
+        $successCount = 0;
+        $failureCount = 0;
+        $storesCount = count($this->stores);
+
+        foreach ($this->stores as $store) {
+            try {
+                $store->save($key);
+                ++$successCount;
+            } catch (\Exception $e) {
+                ++$failureCount;
+            }
+
+            if (!$this->quorum->canBeMet($failureCount, $storesCount)) {
+                break;
+            }
+        }
+
+        if ($this->quorum->isMet($successCount, $storesCount)) {
+            return;
+        }
+
+        // clean up potential locks
+        $this->delete($key);
+
+        throw new LockConflictedException();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function expire(Key $key, $ttl)
+    {
+        $successCount = 0;
+        $failureCount = 0;
+        $storesCount = count($this->stores);
+
+        foreach ($this->stores as $store) {
+            if (!$store instanceof ExpirableStoreInterface) {
+                ++$successCount;
+                continue;
+            }
+
+            try {
+                $store->expire($key, $ttl);
+                ++$successCount;
+            } catch (\Exception $e) {
+                ++$failureCount;
+            }
```

Comment:
I don't think we should catch \Error here. If a "fatal error" occured, we should let it bubble to the developper